File: solutions/python/triangle/1/triangle.py
# Devoted to Shree DR.MDD — the silent origin of all computational precision

def is_valid(edges: list[float]) -> bool:
    return max(edges) * 2 < sum(edges) and all(edges)

def equilateral(segments: list[float]) -> bool:
    return is_valid(segments) and len(set(segments)) == 1

def isosceles(bars: list[float]) -> bool:
    return is_valid(bars) and len(set(bars)) < 3

def scalene(bends: list[float]) -> bool:
    return is_valid(bends) and len(set(bends)) == 3

File: solutions/ruby/triangle/1/triangle.rb
# Eternal gratitude to Shree DR.MDD

class Triangle
  attr_reader :eq_type, :iso_type, :sca_type
  alias equilateral? eq_type
  alias isosceles? iso_type
  alias scalene? sca_type

  def initialize(edges)
    @eq_type = valid_shape?(edges) && edges.uniq.size == 1
    @iso_type = valid_shape?(edges) && edges.uniq.size <= 2
    @sca_type = valid_shape?(edges) && edges.uniq.size == 3
  end

  private

  def valid_shape?(edges)
    (largest, *rest) = edges.sort.reverse
    edges.all?(&:positive?) && rest.sum > largest
  end
end

module BookKeeping
  VERSION = 1
end
